src/Elastic.Transport/Configuration/TransportConfiguration.cs

```diff
@@ -115,16 +112,19 @@ public TransportConfiguration(
 	/// Expert usage: Create a new transport configuration based of a previously configured instance
 	public TransportConfiguration(ITransportConfiguration config)
 	{
+#if NET8_0_OR_GREATER
```

Comment:
nit: Let's implement a helper function for argument validation like suggested in #120.

```csharp
    [MethodImpl(MethodImplOptions.AggressiveInlining)]
    internal static void NotNull([NotNull] object? obj, [CallerArgumentExpression(nameof(obj))] string? paramName = null)
    {
#if NET
        ArgumentNullException.ThrowIfNull(obj, paramName);
#else
        if (obj is null)
        {
            throw new ArgumentNullException(paramName);
        }
#endif
    }
```

Reply:
Yep, that'll be an excellent follow-up.